### PR TITLE
Copy token to clipboard

### PR DESCRIPTION
**Pull request by themsaid**
If you click on the textarea it gets selected and copied, same if you click on the button.

screen shot 2016-09-29 at 9 53 53 pm

### CODE REVIEW

File: resources/views/settings/api/create-token.blade.php [hunk 1]
```diff
@@ -92,11 +92,14 @@
                             at any time from your API settings.
                         </div>
 
-                        <pre><code id="api-token">@{{ showingToken }}</code></pre>
+                        <textarea id="api-token" class="form-control"
```
Comment (morloderex) [inline resources/views/settings/api/create-token.blade.php:95]: It could be like `v-el:apiToken`

**Comment by freddyheppell** [comment]
[Caniuse](http://caniuse.com/#feat=clipboard) says 83% global support for this method, is that enough?

File: resources/assets/js/settings/api/create-token.js [hunk 2]
```diff
@@ -105,6 +105,16 @@ module.exports = {
             $('#modal-show-token').modal('show');
         },
 
+        
+        /**
+         * Select the token and copy to Clipboard.
+         */
+        selectToken() {
+            $('#api-token').select();
```
Comment (morloderex) [inline resources/assets/js/settings/api/create-token.js:120]: Then you could do `$(this.$els.apiToken)` within the call. This way we could get rid of the id?

**Review by morloderex — CHANGES_REQUESTED** [review]
Why not use vue's `v-el` functionality for this functionality this way we could get rid of the need for the id on the element.

**Comment by themsaid** [comment]
@freddinator I'd say yes, it's not a major functionality, if it doesn't work on some old browser the user can just copy manually.

Any better ideas?

**Comment by taylorotwell** [comment]
Is there any way to know if its available? If it's not we could just have the button say "Select All" or something?

**Comment by freddyheppell** [comment]
`document.queryCommandSupported("copy");` would work, so the copy button should be shown if that's true.

**Comment by themsaid** [comment]
@freddinator Thank you :)

**Comment by morloderex** [comment]
@themsaid what about my minor code improvement? :)

**Comment by themsaid** [comment]
@morloderex not sure if it really improves something :)

**Comment by morloderex** [comment]
@themsaid It is a minor improvement.. But what if somebody where to change the id? :)

**Comment by themsaid** [comment]
Updated with the check for copy command support.

**Pull request merged**